week1/project/ecommerce/src/components/ProductList.jsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import products from '../fake-data/all-products'
+import ProductCard from './ProductCard'
+
+const ProductList = ({ selectedCategory, setSelectedCategory }) => {
+  console.log(selectedCategory)
+  return (
+    <div>
+      <ul className="products">
+        {products
+          .filter(
+            (filtredItem) =>
+              filtredItem.category === selectedCategory.slice(6) ||
+              !selectedCategory
+          )
+          .map((item) => (
+            <ProductCard item={item} />
```

Comment:
```suggestion
            <ProductCard item={item} key={item.id} />
```
The key prop needs to be placed on the outermost element returned by the .map() function. In this case, that means adding key to the <ProductCard /> component itself.
![Screenshot 2025-06-29 at 05 07 51](https://github.com/user-attachments/assets/5a703496-19fd-4452-9b4d-0308df1c1866)